### PR TITLE
Ensure span ends

### PR DESCRIPTION
**Pull request by terencechain**
There spans don't call end on it along all control-flow paths. Thus, its associated Context can be leaked. An attacker could exploit this leak for denial of service

### CODE REVIEW

File: beacon-chain/sync/pending_blocks_queue.go [hunk 1]
```diff
@@ -59,6 +59,7 @@ func (s *Service) processPendingBlocks(ctx context.Context) error {
 	randGen := rand.NewGenerator()
 	for _, slot := range slots {
 		ctx, span := trace.StartSpan(ctx, "processPendingBlocks.InnerLoop")
+		defer span.End()
```
Comment (prestonvanloon) [inline beacon-chain/sync/pending_blocks_queue.go:62]: Won't this end up calling span.End() multiple times? The defer only happens when the function returns, not when the loop ends.

One way to fix this is to abstract the inner logic to its own function.

File: validator/client/runner.go [hunk 2]
```diff
@@ -83,6 +83,7 @@ func run(ctx context.Context, v Validator) {
 	}
 	for {
 		ctx, span := trace.StartSpan(ctx, "validator.processSlot")
+		defer span.End()
```
Comment (prestonvanloon) [inline validator/client/runner.go:86]: Same comment here.

File: beacon-chain/sync/pending_blocks_queue.go (hunk 1 above)
Comment (terencechain) [reply beacon-chain/sync/pending_blocks_queue.go:62]: That's what i was afraid of, thanks for confirming. I'll fix it in the draft

Comment (terencechain) [reply beacon-chain/sync/pending_blocks_queue.go]: As it turns out, we already took care of `span.End()` individually on `return` and `continue`. Although we did miss a few spots so I added them. Thanks

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**